### ISSUE TITLE
Extension ChallengePassword is not able to encode Printable String

### ISSUE THREAD

**Issue by xipki** [cose-wg/CBOR-certificates#186]
In draft -09, the extension ChallengePassword allows only UTF-8 string.
Since in RFC 2986, the PrintableString is preffered (see the text below),
the `challegePassword` shall be extended to accept PrintableString.

As described in RFC2986:

```
   A challenge-password attribute must have a single attribute value.

   ChallengePassword attribute values generated in accordance with this
   version of this document SHOULD use the PrintableString encoding
   whenever possible.  If internationalization issues make this
   impossible, the UTF8String alternative SHOULD be used.  PKCS #9-
   attribute processing systems MUST be able to recognize and process
   all string types in DirectoryString values.
```

And the following text (in Page 19, related to ChallengePassword)
```
 ... except if the text string contains
   only the symbols '0'–'9' or 'a'–'f', in which case it is encoded as a
   CBOR byte string
```

shall be limited to **>= 2 and even number of symbols**.

**Comment by xipki** [comment]
One idea is to use the criticality of extension:
- critical: for PRINTABLE STRING
- non-critical: for UTF8 STRING.

**Comment by highlunder** [comment]
Anyone who is up for suggesting a solution in a PR?

**Comment by highlunder** [comment]
Addressed by PR #206
